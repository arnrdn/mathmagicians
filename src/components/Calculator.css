main {
  height: 90vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.interface {
  border: 1px solid #bababa;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  width: 450px;
}

.interface button {
  border: 1px solid #bababa;
  padding: 1.5rem;
}

.orange-btn {
  background-color: #f5913e;
}

.result {
  padding: 2rem;
  background-color: #858694;
  color: #fff;
  text-align: right;
  grid-column: 1/ span 4;
}

.zero {
  grid-column: 1/ span 2;
}
