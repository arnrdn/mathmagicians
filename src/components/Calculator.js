import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: 0, next: null, operation: null });

  const handleClick = (e) => {
    setState((state) => calculate(state, e.target.textContent));
  };

  return (
    <main>
      <div className="interface">
        <div className="result">
          {state.total}
          {state.operation}
          {state.next}
        </div>
        <button type="button" onClick={handleClick}>AC</button>
        <button type="button" onClick={handleClick}>+/-</button>
        <button type="button" onClick={handleClick}>%</button>
        <button type="button" className="orange-btn" onClick={handleClick}>รท</button>
        <button type="button" onClick={handleClick}>7</button>
        <button type="button" onClick={handleClick}>8</button>
        <button type="button" onClick={handleClick}>9</button>
        <button type="button" className="orange-btn" onClick={handleClick}>x</button>
        <button type="button" onClick={handleClick}>4</button>
        <button type="button" onClick={handleClick}>5</button>
        <button type="button" onClick={handleClick}>6</button>
        <button type="button" className="orange-btn" onClick={handleClick}>-</button>
        <button type="button" onClick={handleClick}>1</button>
        <button type="button" onClick={handleClick}>2</button>
        <button type="button" onClick={handleClick}>3</button>
        <button type="button" className="orange-btn" onClick={handleClick}>+</button>
        <button type="button" className="zero" onClick={handleClick}>0</button>
        <button type="button" onClick={handleClick}>.</button>
        <button type="button" className="orange-btn" onClick={handleClick}>=</button>
      </div>
    </main>
  );
};

export default Calculator;
