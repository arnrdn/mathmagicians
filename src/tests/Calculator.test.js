import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('Match the Calculator component snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();

  expect(tree).toMatchSnapshot();
});
