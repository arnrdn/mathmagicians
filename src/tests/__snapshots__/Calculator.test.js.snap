// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the Calculator component snapshot 1`] = `
<div
  className="content-container"
>
  <div
    className="title-calc"
  >
    <h2>
      Let's do some math!
    </h2>
    <div
      className="interface"
    >
      <div
        className="result"
      >
        0
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
